XenAdmin/Wizards/PatchingWizard/AutomatedUpdatesBasePage.cs

```diff
@@ -174,8 +174,7 @@ private bool StartUpgradeWorkers()
                 {
                     //add a revert pre-check action for this pool
                     var curPool = pool;
-                    var problemsToRevert = PrecheckProblemsActuallyResolved.Where(a =>
-                        a.SolutionAction != null && Helpers.GetPoolOfOne(a.SolutionAction.Connection).Equals(curPool)).ToList();
+                    var problemsToRevert = PrecheckProblemsActuallyResolved.Where(a => Helpers.GetPoolOfOne(a.SolutionAction?.Connection).Equals(curPool)).ToList();

```

Comment:
That's not what I wrote earlier, I wrote: `curPool.Equals(Helpers.GetPoolOfOne(a.SolutionAction?.Connection))`. `null.Equals(curPool)` will crash, but `curPool.Equals(null)` will not.

Reply:
oops! my bad